src/yz_wm_index.erl

```diff
@@ -256,14 +270,29 @@ malformed_request(RD, S) ->
           end
     end.
 
-%% Returns a 409 Conflict if this index already exists
-is_conflict(RD, S) when S#ctx.method =:= 'PUT' ->
-    IndexName = S#ctx.index_name,
-    case exists(IndexName) of
-        true  ->
-            {true, RD, S};
-        false ->
-            {false, RD, S}
+-spec malformed_timeout_param(#wm_reqdata{}, context()) ->
+    {boolean(), #wm_reqdata{}, context()}.
+%% @doc Check that the timeout parameter is are a
+%%      string-encoded integer.  Store the integer value
+%%      in context() if so.
+malformed_timeout_param(RD, S) ->
+    case wrq:get_qs_value("timeout", none, RD) of
+        none ->
+            {false, RD, S};
+        TimeoutStr ->
+            try
+                Timeout = list_to_integer(TimeoutStr),
+                {false, RD, S#ctx{timeout=Timeout}}
+            catch
+                _:_ ->
+                    {true,
+                     wrq:append_to_resp_body(io_lib:format("Bad timeout "
```

Comment:
Can this indicate that the reason the timeout is bad is that it can't be interpreted as an integer?